cmd/utils.go

```diff
@@ -95,9 +96,92 @@ func splitStr(path, sep string, n int) []string {
 	return splits
 }
 
+// Parse scheme format, supported formats are
+// - "http"
+// - "http+s3v2"
+// - "http+s3v4"
+func parseScheme(scheme string) (string, string, *probe.Error) {
+	tokens := strings.Split(scheme, "+")
+	if len(tokens) > 2 || len(tokens) < 1 {
+		return "", "", errInvalidArgument().Trace(scheme)
+	}
+
+	httpScheme := tokens[0]
+	if httpScheme != "http" && httpScheme != "https" {
+		return "", "", errInvalidArgument().Trace(scheme, httpScheme)
+	}
+
+	signatureType := "s3v4"
+	if len(tokens) == 2 {
+		signatureType = tokens[1]
+	}
+	if !isValidAPI(signatureType) {
+		return "", "", errInvalidArgument().Trace(signatureType, scheme)
+	}
+
+	return tokens[0], signatureType, nil
+}
+
+func buildS3ConfigFromEnv(envConfig, urlStr string) (*Config, *probe.Error) {
+	_, p := url2Alias(urlStr)
```

Comment:
Can you move this line just above where 'p' variable is used ?

Reply:
Sure